### DISCUSSION TITLE
Can I drop edge from toolbar?

### DISCUSSION THREAD

**Discussion by reuse-ay** [maxGraph/maxGraph#414] [Q&A] [answered]
I'm trying to drop from toolbar by reference to toolbar [sample on storybook](https://maxgraph.github.io/maxGraph/demo/?path=/story/toolbars-toolbar--default).
I defined function `addEdge` like `addVertex` in sample. The difference is only using `setEdge` or `setVertex` in function. I can drop vertex. However, I can't drop edge. What can I do? 

[reproduction:](https://stackblitz.com/~/github.com/reuse-ay/angular_practice?file=src/app/content/toolbar/toolbar.component.ts)

you can see from "MaxGraph Toolbar" in upper left menu.

**Comment by tbouffard** [comment, marked as answer]
Yes this is possible, see #416

You must set the sourcePoint and targetPoint properties in the edge geometry, otherwise the edge is not displayed (maxGraph is unable to position the edge)

**Reply by reuse-ay** [reply]
Thanks! I successfully implemented.
[sample](https://maxgraph.github.io/maxGraph/demo/?path=/story/toolbars-toolbar--default)
